lib/improver/cli/nowcast_extrapolate.py

```diff
@@ -219,85 +160,34 @@ def process(input_cube, u_cube, v_cube, speed_cube, direction_cube,
-            calculate these accumulations. This interval must be a factor of
-            the lead_time_interval.
-            Default is 0.
-        accumulation_period (int):
-            The period over which the accumulation is calculated (mins).
-            Only full accumulation periods will be computed. At lead times
-            that are shorter than the accumulation period, no accumulation
-            output will be produced.
-        accumulation_units (str):
-            Desired units in which the accumulations should be expressed.
-            e.g. 'mm'
-            Default is 'm'.
 
     Returns:
-        (tuple): tuple containing:
-            **accumulation_cubes** (iris.cube.CubeList):
-                A cubelist containing precipitation accumulation cubes where
-                the accumulation periods are determined by the
-                lead_time_interval.
-            **forecast_to_return** (iris.cube.CubeList):
-                New cubes with updated time and extrapolated data.
+        iris.cube.CubeList:
+            New cubes with updated time and extrapolated data.
 
     Raises:
         ValueError:
-            can either use s_cube and d_cube or u_cube and v_cube.
-            Therefore: (s and d)⊕(u and v)
-        ValueError:
-            If accumulation_fidelity is greater than 0 and max_lead_time is not
-            cleanly divisible by accumulation_fidelity.
+            can either use speed_cube and direction_cube or u_cube and v_cube.
     """
-    if (speed_cube and direction_cube) and not (u_cube or v_cube):
-        u_cube, v_cube = ResolveWindComponents().process(
-            speed_cube, direction_cube)
-    elif (u_cube or v_cube) and (speed_cube or direction_cube):
+    if (speed_cube or direction_cube) and (u_cube or v_cube):
         raise ValueError('Cannot mix advection component velocities with speed'
                          ' and direction')
+    if not (speed_cube and direction_cube) and not (u_cube and v_cube):
+        raise ValueError('Either speed and direction or u and v cubes '
+                         'are needed.')
 
-    # determine whether accumulations are also to be returned, and modify time
-    # interval if finer intervals are needed for accumulations
-    time_interval = lead_time_interval
-    if accumulation_fidelity > 0:
-        fraction, _ = np.modf(max_lead_time / accumulation_fidelity)
-        if fraction != 0:
-            msg = ("The specified lead_time_interval ({}) is not cleanly "
-                   "divisible by the specified accumulation_fidelity ({}). As "
-                   "a result the lead_time_interval cannot be constructed from"
-                   " accumulation cubes at this fidelity.")
-            raise ValueError(msg.format(lead_time_interval,
-                                        accumulation_fidelity))
-        time_interval = accumulation_fidelity
+    if (speed_cube and direction_cube) and not (u_cube or v_cube):
```

Comment:
Do the previous `if` tests mean that this `if` test can be reduced to `if speed_cube and direction_cube`? As far as I can tell, I would have thought that, as a result of your previous tests, you already know that if `speed_cube` and `direction_cube` are not None, then `u_cube` and `v_cube` will be None.

Reply:
fixed